prov/util/src/util_attr.c

```diff
@@ -230,11 +231,23 @@ static int ofi_info_to_util(uint32_t version, const struct fi_provider *prov,
 	if (ofi_dup_addr(core_info, *util_info))
 		goto err;
 
-	(*util_info)->domain_attr->name = strdup(core_info->domain_attr->name);
-	if (!(*util_info)->domain_attr->name) {
-		FI_WARN(prov, FI_LOG_FABRIC,
-			"Unable to allocate domain name\n");
-		goto err;
+	/* Release 1.4 brought standardized domain names across IP based
+	 * providers. Before this release, the usNIC provider would return a
+	 * NULL domain name from fi_getinfo. For compatibility reasons, allow a
+	 * NULL domain name when apps are requesting version < 1.4.
+	 */
+	if (FI_VERSION_GE(version, FI_VERSION(1, 4)))
+		assert(core_info->domain_attr->name);
```

Comment:
You can move the if check into the assert.  E.g. assert(FI_VERSION_LT(1.4) || name)